### PR TITLE
PWGHF: tasks ToXiPi - selections QA, tree creator

### PR DESCRIPTION
**Pull request by ZFederica**
- add tree creator
- add temporary histograms for QA of selections

### CODE REVIEW

**Review by lvermunt — APPROVED** [review]

File: PWGHF/TableProducer/treeCreatorToXiPi.cxx [hunk 1]
```diff
@@ -0,0 +1,327 @@
...
+///        In this file are defined and filled the output tables
+///
+/// \author Federica Zanone <federica.zanone@cern.ch>, Heidelberg University & GSI
+
+#include "Framework/runDataProcessing.h"
+#include "Framework/AnalysisTask.h"
+#include "PWGHF/DataModel/CandidateReconstructionTables.h"
+#include "PWGHF/DataModel/CandidateSelectionTables.h"
+#include "Common/Core/trackUtilities.h"
+#include "ReconstructionDataFormats/DCA.h"
```
Comment (vkucera) [inline PWGHF/TableProducer/treeCreatorToXiPi.cxx:23]: Sort

File: PWGHF/TableProducer/treeCreatorToXiPi.cxx [hunk 2]
```diff
@@ -0,0 +1,327 @@
...
+} // namespace o2::aod
+
+/// Writes the full information in an output TTree
+struct HfTreeCreatorToXiPi {
+
+  Produces<o2::aod::HfToXiPiFull> rowCandidateFull;
+  Produces<o2::aod::HfToXiPiEvents> rowCandidateEvents;
+
+  Configurable<bool> doNotMc{"doNotMc", true, "Process data tree writer"};
+  Configurable<bool> doMc{"doMc", false, "Process MC tree writer"};
```
Comment (vkucera) [inline PWGHF/TableProducer/treeCreatorToXiPi.cxx:173]: These are not needed. Each process function has its own configurable.

File: PWGHF/TableProducer/treeCreatorToXiPi.cxx [hunk 3]
```diff
@@ -0,0 +1,327 @@
...
+    // Filling event properties
+    rowCandidateEvents.reserve(collisions.size());
+    for (auto& collision : collisions) {
+      fillEvent(collision);
+    }
+
+    // Filling candidate properties
+    rowCandidateFull.reserve(candidates.size());
+    for (auto& candidate : candidates) {
+      fillCandidate(candidate, -2, -2);
```
Comment (vkucera) [inline PWGHF/TableProducer/treeCreatorToXiPi.cxx:299]: Why -2? It corresponds to matched anti Omega_c.

File: PWGHF/TableProducer/treeCreatorToXiPi.cxx [hunk 4]
```diff
@@ -0,0 +1,327 @@
...
+
+    // Filling event properties
+    rowCandidateEvents.reserve(collisions.size());
+    for (auto& collision : collisions) {
+      fillEvent(collision);
+    }
+
+    // Filling candidate properties
+    rowCandidateFull.reserve(candidates.size());
+    for (auto& candidate : candidates) {
```
Comment (vkucera) [inline PWGHF/TableProducer/treeCreatorToXiPi.cxx:298]:
```suggestion
    for (auto const& candidate : candidates) {
```

File: PWGHF/TableProducer/treeCreatorToXiPi.cxx [hunk 5]
```diff
@@ -0,0 +1,327 @@
...
+      debugMc);
+  }
+
+  void processData(aod::Collisions const& collisions,
+                   soa::Join<aod::HfCandToXiPi, aod::HfSelToXiPi> const& candidates)
+  {
+
+    // Filling event properties
+    rowCandidateEvents.reserve(collisions.size());
+    for (auto& collision : collisions) {
```
Comment (vkucera) [inline PWGHF/TableProducer/treeCreatorToXiPi.cxx:292]:
```suggestion
    for (auto const& collision : collisions) {
```

File: PWGHF/TableProducer/treeCreatorToXiPi.cxx [hunk 6]
```diff
@@ -0,0 +1,327 @@
...
+  }
+  PROCESS_SWITCH(HfTreeCreatorToXiPi, processData, "Process data tree writer", doNotMc);
+
+  void processMc(aod::Collisions const& collisions,
+                 soa::Join<aod::HfCandToXiPi, aod::HfSelToXiPi, aod::HfToXiPiMCRec> const& candidates)
+  {
+
+    // Filling event properties
+    rowCandidateEvents.reserve(collisions.size());
+    for (auto& collision : collisions) {
```
Comment (vkucera) [inline PWGHF/TableProducer/treeCreatorToXiPi.cxx:310]:
```suggestion
    for (auto const& collision : collisions) {
```

File: PWGHF/TableProducer/treeCreatorToXiPi.cxx [hunk 7]
```diff
@@ -0,0 +1,327 @@
...
+      fillEvent(collision);
+    }
+
+    // Filling candidate properties
+    rowCandidateFull.reserve(candidates.size());
+    for (auto& candidate : candidates) {
```
Comment (vkucera) [inline PWGHF/TableProducer/treeCreatorToXiPi.cxx:316]:
```suggestion
    for (auto const& candidate : candidates) {
```

**Review by vkucera — APPROVED** [review]

**Review by vkucera — CHANGES_REQUESTED** [review]
Please see my comments.

File: PWGHF/TableProducer/treeCreatorToXiPi.cxx (hunk 1 above)
Comment (ZFederica) [reply PWGHF/TableProducer/treeCreatorToXiPi.cxx:23]: Ok

File: PWGHF/TableProducer/treeCreatorToXiPi.cxx (hunk 2 above)
Comment (ZFederica) [reply PWGHF/TableProducer/treeCreatorToXiPi.cxx:173]: Ok

File: PWGHF/TableProducer/treeCreatorToXiPi.cxx (hunk 4 above)
Comment (ZFederica) [reply PWGHF/TableProducer/treeCreatorToXiPi.cxx:298]: Ok

File: PWGHF/TableProducer/treeCreatorToXiPi.cxx (hunk 5 above)
Comment (ZFederica) [reply PWGHF/TableProducer/treeCreatorToXiPi.cxx:292]: Ok

**Comment by ZFederica** [comment]
@vkucera all the changes are implemented

**Review by vkucera — APPROVED** [review]

**Comment by vkucera** [comment]
> @vkucera all the changes are implemented

Thanks @ZFederica

**Pull request merged**